### ISSUE TITLE
Report use of `instanceof`

### ISSUE THREAD

**Issue by brettz9** [eslint/eslint#11132]
**Please describe what the rule should do:**

Report use of `instanceof`. Ideally, the rule would not be enforced if detecting that the class being tested possessed [`Symbol.hasInstance`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/hasInstance).

**What category of rule is this? (place an "X" next to just one item)**

[x] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
x instanceof y
```

**Why should this rule be included in ESLint (instead of a plugin)?**

`instanceof`--unless used against a class with`Symbol.hasInstance`--is not safe across frames and potentially problematic for any code, including library code, used across frames.

**Are you willing to submit a pull request to implement this rule?**

Not at this time.

**Comment by nzakas** [comment]
Thanks for the suggestion. We're generally not accepting new rules that simply disallow a particular syntax because you can disallow any syntax you want using `no-restricted-syntax`.

If you really want a standalone rule, I'd suggest creating your own and distributing it via a plugin.
